stockflow/stockflow_schema.json

```diff
@@ -0,0 +1,413 @@
+        },
+        "description": {
+          "type": "string"
+        },
+        "model_version": {
+          "type": "string"
+        }
+      },
+      "required": ["name", "schema", "description"]
+    },
+    "properties": {
+      "type": "object"
+    },
+    "model": {
+      "type": "object",
+      "properties": {
+        "stocks": {
+          "$ref": "#/$defs/stocks"
+        },
+        "flows": {
+          "$ref": "#/$defs/flows"
+        },
+        "links": {
+          "$ref": "#/$defs/links"
+        },
+        "auxiliaries": {
+          "$ref": "#/$defs/auxiliaries"
+        }
+        },
+      "required": ["stocks", "flows", "links"],
+      "additionalProperties": false
+    },
+    "semantics": {
+      "type": "object",
+      "description": "Information specific to a given semantics (e.g., ODEs) associated with a model.",
+      "properties": {
+        "ode": {
+          "$ref": "#/$defs/odeSemantics"
+        },
+        "typing": {
+          "description": "(Optional) Information for aligning models for stratification",
+          "$ref": "#/$defs/typingSemantics"
+        },
+        "span": {
+          "type": "array",
+          "description": "(Optional) Legs of a span, each of which are a full ASKEM Petri Net",
+          "items": {
+            "type": "object",
+            "$ref": "#/$defs/typingSemantics"
+          }
+        }
+      }
+    },
+    "metadata": {
+      "type": "object",
+      "description": "(Optional) Information not useful for execution of the model, but that may be useful to some consumer in the future. E.g. creation timestamp or source paper's author."
+    }
+  },
+  "$defs": {
+    "stocks": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "id": {
+            "type": "string"
+          },
+          "name": {
+            "type": "string"
+          },
+          "description": {
+            "type": "string"
+          },
+          "grounding": {
+            "$ref": "#/$defs/grounding"
+          },
+          "units": {
+            "$ref": "#/$defs/unit"
+          }
+        },
+        "required": ["id"]
+      }
+    },
+    "flows": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "id": {
+            "type": "string"
+          },
+          "upstream_stock": {
+            "type": "string"
+          },
+          "downstream_stock": {
+            "type": "string"
+          },
+          "rate_expression": {
+            "type": "string"
+          },
+          "rate_expression_mathml": {
+            "type": "string"
+          },
+          "grounding": {
+            "$ref": "#/$defs/grounding"
+          },
+          "properties": {
+            "$ref": "#/$defs/properties"
+          }
+        },
+        "required": ["id", "upstream_stock", "downstream_stock"]
+      }
+    },
+    "links": {
+      "type": "array",
```

Comment:
I can foresee us needed to include additional information on the links, such as "color"/"category", "polarity", "label", etc.
I am not sure which, if any, of these should be included in the schema or if we should just include a `meta`/`extras` property to store any of this extra information.

Reply:
I would put any of those keys under an optional `properties` block.